### PR TITLE
Updated system modes roadmap

### PR DESCRIPTION
**Pull request by norro**
updates system modes documentation, e.g., roadmap

### CODE REVIEW

File: _docs/concepts/client_library/system_modes/index.md [hunk 1]
```diff
@@ -121,13 +141,16 @@ A simple example is provided at:
 
 ### 2020
 
-- Diagnostics framework for micro-ROS, interoperating with ROS 2 diagnostics
```
Comment (ralph-lange) [inline _docs/concepts/client_library/system_modes/index.md:124]: Please add the port of ROS diagnostics to ROS 2 for the years 2019 and 2020 and explain that this was a prerequisite for future micro-ROS diagnostics.

**Review by ralph-lange — CHANGES_REQUESTED** [review]

**Comment by norro** [comment]
Not much more to change, IMHO.

File: _docs/concepts/client_library/system_modes/index.md [hunk 2]
```diff
@@ -116,18 +136,22 @@ A simple example is provided at:
 
 - Extended lifecycle concept and implementation for ROS 2 C++
 - Mode inference and mode manager for ROS 2 C++
-- ROS diagnostics port from ROS 1 to ROS 2
-- Concept for bottom-up system modes rules
+- ROS diagnostics port from ROS 1 to ROS 2 (prerequisite for micro-ROS diagnistics)
```
Comment (ralph-lange) [inline _docs/concepts/client_library/system_modes/index.md:139]:
```suggestion
- ROS diagnostics port from ROS 1 to ROS 2 (prerequisite for micro-ROS diagnostics)
```

File: _docs/concepts/client_library/system_modes/index.md [hunk 3]
```diff
@@ -116,18 +136,22 @@ A simple example is provided at:
...
+- Lightweight concept for error propagation and recovery mechanisms
 
 ### 2020
 
-- Diagnostics framework for micro-ROS, interoperating with ROS 2 diagnostics
-- MCU-specific diagnostics functions for resource usage on RTOS layer, latencies, statistics from middleware, etc.
 - Integration of mode manager with real-time executor and/or roslaunch
-- Lightweight concept for specifying error propagation and recovery mechanisms
+- Lightweight implementation for specifying error propagation and recovery mechanisms
+- ROS diagnostics port from ROS 1 to ROS 2 (prerequisite for micro-ROS diagnistics)
```
Comment (ralph-lange) [inline _docs/concepts/client_library/system_modes/index.md:146]:
```suggestion
- ROS diagnostics port from ROS 1 to ROS 2 (prerequisite for micro-ROS diagnostics)
```

**Review by ralph-lange — CHANGES_REQUESTED** [review]
Just typos. Sorry, I overlooked the existence of the 2019 section with item on ROS 2 diagnostics in my first review. Nevertheless, I think it is a good idea to explain the prerequisite for micro-ROS diagnostics as done now.

**Comment by norro** [comment]
Agreed.
Seems that you have to review once more for me to be able to merge without using the admin overwrite. ;)

**Review by ralph-lange — APPROVED** [review]

**Pull request merged**